File: Leetcode/easy/1TwoSum/twosum.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int>hashMap;

        for(int i = 0 ; i < nums.size() ; i++){
          int compliment = target - nums[i];
          if(hashMap.find(compliment) != hashMap.end()){
            return {hashMap[compliment] , i};
          }
          else{
            hashMap[nums[i]] = i;
          }
        }

        return {};
    }
};
int main() {
    Solution sol;
    
    return 0;
}

File: DSARestart/dailyOne/february/27Feb/twoSum.ts
/// thsi is very easy problem in this problem we need to just find the complement of the the substrrationi from the entire array
// storing complemment also as the 
const twoSum = (nums:number[], target:number):number[]=>{
    const hashMap = new Map()

    for(let i = 0 ; i < nums.length ; i++){
        // compliment 
        const complement = target - nums[i]
        // iff complement exist on hashMap
        if(hashMap.has(complement)){
            return [hashMap.get(complement), i]
        }
        else{
            hashMap.set(nums[i],i )
        }
    }
    return []
}

console.log(twoSum([2,7,11,15], 9))